CMakeLists.txt

```diff
@@ -34,7 +34,11 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-duplicate-decl-specifier \
 cmake_minimum_required(VERSION 3.20 FATAL_ERROR)
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
-set(CHIPSTAR_VERSION 1.1)
+set(CHIPSTAR_MAJOR_VERSION 1)
+set(CHIPSTAR_MINOR_VERSION 1)
+set(CHIPSTAR_PATCH_VERSION 0)
+math(EXPR CHIPSTAR_VERSION "${CHIPSTAR_MAJOR_VERSION}*10000000+${CHIPSTAR_MINOR_VERSION}*100000+${CHIPSTAR_PATCH_VERSION}" OUTPUT_FORMAT DECIMAL)
```

Comment:
A problem with this is that the project version appears as `10100000` instead of `1.1`. An example from <build-dir>/CMakeCache.txt:
```
//Value Computed by CMake
CMAKE_PROJECT_VERSION:STATIC=10100000

//Value Computed by CMake
CMAKE_PROJECT_VERSION_MAJOR:STATIC=10100000

//Value Computed by CMake
CMAKE_PROJECT_VERSION_MINOR:STATIC=
```

Reply:
@linehill 
Not sure why do we need to maintain two different versions (CHIPSTAR_VERSION and CHIPSTAR_VERSION_INT). This will create confusion to others. 
I did not find AMD is having similar two version of macros. Rather having CHIPSTAR_VERSION_INT I would prefer to have it as below

hipError_t hipRuntimeGetVersion(int *RuntimeVersion) {
......
   *RuntimeVersion = CHIPSTAR_MAJOR_VERSION*10000000+CHIPSTAR_MINOR_VERSION*100000+CHIPSTAR_PATCH_VERSION
} 

Let me know if you think other way.

Reply:
> Not sure why do we need to maintain two different versions (CHIPSTAR_VERSION and CHIPSTAR_VERSION_INT). This will create confusion to others.

Fair enough.

> Rather having CHIPSTAR_VERSION_INT I would prefer to have it as below
>
> hipError_t hipRuntimeGetVersion(int *RuntimeVersion) {
> ......
> *RuntimeVersion = CHIPSTAR_MAJOR_VERSION x 10000000+CHIPSTAR_MINOR_VERSION x > 100000+CHIPSTAR_PATCH_VERSION
> }

Sounds good to me.